### PR TITLE
Fix not being able to use an enderpearl in creative mode

### PR DESCRIPTION
**Pull request by Connor-Colenso**

### CODE REVIEW

File: src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@Mixin(ItemEnderPearl.class)
+public abstract class MixinItemEnderPearl extends Item {
+
+    /**
+     * @author Colen
+     * @reason Enables enderpearls to be used in creative by removing check.
+     */
+    @Overwrite
+    public ItemStack onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player) {
```
Comment (mist475) [inline src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java:21]: there's no need to have this as an overwrite. Redirecting the call to player.capabilities.isCreativeMode to always return false is cleaner as that way other mixins can still modify the same method

File: src/main/resources/mixins.hodgepodge.early.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
+
```
Comment (mist475) [inline src/main/resources/mixins.hodgepodge.early.json:1]: unnecessary changes here

File: src/main/resources/mixins.hodgepodge.early.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 {
   "required": true,
   "minVersion": "0.8.3-GTNH",
   "package": "com.mitchej123.hodgepodge.mixins.early",
   "refmap": "mixins.hodgepodge.refmap.json",
   "target": "@env(DEFAULT)",
   "compatibilityLevel": "JAVA_8"
-}
-
+}
```
Comment (mist475) [inline src/main/resources/mixins.hodgepodge.early.json:9]: and here

**Review by mist475 — CHANGES_REQUESTED** [review]

File: src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java (hunk 1 above)
Comment (Connor-Colenso) [reply src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java:21]: I have no idea how one would redirect this specific call within another function. Is there an example to look at?

Comment (mist475) [reply src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java:21]: [Here](https://github.com/GTNewHorizons/Hodgepodge/blob/master/src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinEntityLivingDrop.java#L29) is an example for PUTFIELD, in your case you need GETFIELD but the rest is identical (except that you need a different target and method of course)

Comment (Connor-Colenso) [reply src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java]: @mist475 I think I've done this now, can you double check for me? I haven't done a mixin in this manner before.

Comment (mist475) [reply src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java]: While that works, that might have side effects. What you should do is:

```java
    @Redirect(
            method = "onItemRightClick",
            at = @At(
                    value = "FIELD",
                    target = "Lnet/minecraft/entity/player/PlayerCapabilities;isCreativeMode:Z"))
    public boolean hodgepodge$removeCreativeCHeck(PlayerCapabilities instance) {
        return false;
    }
```

**Review by mist475 — APPROVED** [review]
